### ISSUE TITLE
Stack-overflow (OSS-Fuzz 869)

### ISSUE THREAD

**Issue by nlohmann** [nlohmann/json#517]
```
Detailed report: https://oss-fuzz.com/testcase?key=5116967767769088

Project: json
Fuzzer: libFuzzer_json_parse_cbor_fuzzer
Job Type: libfuzzer_asan_json
Platform Id: linux

Crash Type: Stack-overflow
Crash Address: 0x7ffc0dea4b58
Crash State:
  nlohmann::basic_json<std::__1::map, std::__1::vector, std::__1::basic_string<cha
  
Sanitizer: address (ASAN)

Regressed: https://oss-fuzz.com/revisions?job=libfuzzer_asan_json&range=201701031958:201701032147

Reproducer Testcase: https://oss-fuzz.com/download/AMIfv95GDH2Mys_drMGzg9fUR7UPBBqUF_LhjJm-rgKw6x-oTwvsux765EavyU1j3OwyrVWiw_gW8F_Ia6rk-diY-dnPJ2rD592wN1mKQoik0vlg1QtZCaI2yCI5uBCQuaf1PBLkLbIsMQtTywZ5nmE_AZlrvhk2Mye_GJ4oM8wNGSM96vFsgM_IpJ8RXU_sE1htQ639Zyj55qzACG-LDlRAXFeA-Pxd9X2Os5u259dzhhjmH6eqLlXItedTywfMtygaPkYlc4R05g78-SwjpAyOv0BNR4dx2jkcD9xHXNjjAY-232t-qf9oWf2T66QqT_FoMbm9qtBf1FAI8H6Af4e9PUMVB2xjPOIe8uCQh50tqv58mnKY41ZAwOcstJChSYmBAAFsUSnV?testcase_id=5116967767769088


Issue filed automatically.

See https://github.com/google/oss-fuzz/blob/master/docs/reproducing.md for more information.

This bug is subject to a 90 day disclosure deadline. If 90 days elapse
without an upstream patch, then the bug report will automatically
become visible to the public.
```

[clusterfuzz-testcase-5116967767769088.zip](https://github.com/nlohmann/json/files/851775/clusterfuzz-testcase-5116967767769088.zip)

**Comment by nlohmann** [comment]
This seems to be a classical stack overflow due to excessive nesting of arrays.

**Comment by nlohmann** [comment]
ClusterFuzz has detected this issue as fixed in range 201705051619:201705061619.
